DBConnect.cc

```diff
@@ -54,4 +58,310 @@ void ConnectToMongoDB() {
+    const char* uri_string = "mongodb://localhost:27017";
+    mongoc_uri_t* uri;
+    mongoc_client_t* client;
+    mongoc_database_t* database;
+    mongoc_collection_t* collection;
+    bson_t *command, reply, *insert, layout, *query;
+    bson_error_t error;
+    bool retval;
+    char* str;
+    char user[50];
+    bool result = true;
+
+    mongoc_init();
+
+    uri = mongoc_uri_new_with_error(uri_string, &error);
+    if (!uri) {
+        fprintf(stderr,
+            "failed to parse URI: %s\n"
+            "error message:       %s\n",
+            uri_string, error.message);
+        return EXIT_FAILURE;
+    }
+
+    client = mongoc_client_new_from_uri(uri);
+    if (!client) {
+        return EXIT_FAILURE;
+    }
+
+    mongoc_client_set_appname(client, "carta_backend");
+
+    database = mongoc_client_get_database(client, "CARTA");
+    collection = mongoc_client_get_collection(client, "CARTA", "layouts");
+
+    command = BCON_NEW("ping", BCON_INT32(1));
+    retval = mongoc_client_command_simple(client, "admin", command, NULL, &reply, &error);
+    if (!retval) {
+        fprintf(stderr, "%s\n", error.message);
+        return EXIT_FAILURE;
+    }
+
+    str = bson_as_json(&reply, NULL);
+
+    cuserid(user);
+
+    insert = bson_new();
+
+    bson_init(&layout);
+    BSON_APPEND_UTF8(&layout, "username", user);
+    BSON_APPEND_UTF8(&layout, "name", name.c_str());
+    
+    if( !strcmp(json_string.c_str(), "") ) {
+      if (!mongoc_collection_delete_one(collection, &layout, NULL, NULL, &error)) {
+                fprintf(stderr, "Delete failed: %s\n", error.message);
+                result = false;
+      }
+    } else {
+      BSON_APPEND_UTF8(&layout, "json_string", json_string.c_str());
+
+      if (!mongoc_collection_insert_one(collection, &layout, NULL, NULL, &error)) {
+        fprintf(stderr, "%s\n", error.message);
+        result = false;
+      }
+    }
+    bson_destroy(insert);
+    bson_destroy(&reply);
+    bson_destroy(command);
+    bson_free (str);
+
+    mongoc_collection_destroy(collection);
+    mongoc_database_destroy(database);
+    mongoc_uri_destroy(uri);
+    mongoc_client_destroy(client);
+    mongoc_cleanup();
+
+    return result;
+}
+
+//
+bool GetLayoutsFromDB(CARTA::RegisterViewerAck* ack_message_ptr) {
+    const char* uri_string = "mongodb://localhost:27017";
+    mongoc_uri_t* uri;
+    mongoc_client_t* client;
+    mongoc_database_t* database;
+    mongoc_collection_t* collection;
+    mongoc_cursor_t* cursor;
+    bson_t *command, reply, *insert, userconf, *query;
+    const bson_t* doc;
+    bson_error_t error;
+    char* str;
+    char* uvalue = NULL;
+    bool retval;
+    char user[50];
+
+    mongoc_init();
+
+    uri = mongoc_uri_new_with_error(uri_string, &error);
+    if (!uri) {
+        fprintf(stderr,
+            "failed to parse URI: %s\n"
+            "error message:       %s\n",
+            uri_string, error.message);
+        return false;
```

Comment:
We should consistently use `EXIT_FAILURE`.

Reply:
I'm retracting this; it was a problem when it was an inconsistency, but now that we're using true/false consistently that is also fine.